core/src/Kokkos_Macros.hpp

```diff
@@ -502,21 +502,9 @@
 
 #define KOKKOS_IMPL_CTOR_DEFAULT_ARG KOKKOS_INVALID_INDEX
 
-#if (defined(KOKKOS_ENABLE_CXX14) || defined(KOKKOS_ENABLE_CXX17) || \
-     defined(KOKKOS_ENABLE_CXX20))
 #define KOKKOS_CONSTEXPR_14 constexpr
```

Comment:
I cannot think of any good reason to keep this macro around.

Reply:
What about user code? This is not marked `INTERNAL` or `IMPL` or anything like that.